plugins/ptp_operator/metrics/metrics.go

```diff
@@ -0,0 +1,421 @@
+		glog.Warningf("no delay from master process %s out of sync", processName)
+	}
+
+	return
+}
+
+func extractRegularMetrics(processName, output string) (offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64, clockState ceevent.SyncState) {
+	// remove everything before the rms string
+	// This makes the out to equals
+	output = strings.Replace(output, "path", "", 1)
+	index := strings.Index(output, "offset")
+	output = output[index:]
+	fields := strings.Fields(output)
+
+	if len(fields) < 5 {
+		glog.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return
+	}
+
+	offsetFromMaster, err := strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	maxOffsetFromMaster, err = strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse max offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	frequencyAdjustment, err = strconv.ParseFloat(fields[4], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse frequency adjustment output %s error %v", processName, fields[4], err)
+	}
+
+	state := fields[2]
+
+	switch state {
+	case unLocked:
+		clockState = ceevent.FREERUN
+	case clockStep:
+		clockState = ceevent.FREERUN
+	case locked:
+		clockState = ceevent.LOCKED
+	default:
+		glog.Errorf("%s -  failed to parse clock state output `%s` ", processName, fields[2])
+	}
+
+	if len(fields) >= 7 {
+		delayFromMaster, err = strconv.ParseFloat(fields[6], 64)
+		if err != nil {
+			glog.Errorf("%s failed to parse delay from master output %s error %v", processName, fields[6], err)
+		}
+	} else {
+		// If there is no delay from master this mean we are out of sync
+		glog.Warningf("no delay from master process %s out of sync", processName)
+		clockState = ceevent.HOLDOVER
+	}
+
+	return
+}
+
+// ExtractPTP4lEvent ... extract event form ptp4l logs
+func extractPTP4lEventState(processName, output string) (clockState ceevent.SyncState) {
+	// This makes the out to equals
+	//phc2sys[187248.740]:[ens5f0] CLOCK_REALTIME phc offset        12 s2 freq   +6879 delay    49
+	// phc2sys[187248.740]:[ens5f1] CLOCK_REALTIME phc offset        12 s2 freq   +6879 delay    49
+	//ptp4l[3535499.740]: [ens5f0] master offset         -6 s2 freq   -1879 path delay        88
+
+	/*
+		"INITIALIZING to LISTENING on INIT_COMPLETE"
+		"INITIALIZING to LISTENING on INIT_COMPLETE"
+		"LISTENING to UNCALIBRATED on RS_SLAVE"
+		"UNCALIBRATED to SLAVE on MASTER_CLOCK_SELECTED"
+		"SLAVE to FAULTY on FAULT_DETECTED"
+		"LISTENING to FAULTY on FAULT_DETECTED"
+		"FAULTY to LISTENING on INIT_COMPLETE"
+		"FAULTY to SLAVE on INIT_COMPLETE"
+		"SLAVE to UNCALIBRATED on SYNCHRONIZATION_FAULT"
+	*/
+	var rePTPStatus = regexp.MustCompile(`INITIALIZING|LISTENING|UNCALIBRATED|FAULTY|FAULT_DETECTED|SYNCHRONIZATION_FAULT`)
```

Comment:
patterns like `FAULTY` looks too general to me.

Reply:
That's a fixed set of values coming from the PTP logs.  All values will transition to FREERUN. After testing in real PTP  labs if we see the difference then we can enhance it. Also, I have to add a configurable threshold variable.